### DISCUSSION TITLE
How to track errors & Exceptions from Twitter API if i'm using twittered client. ?

### DISCUSSION THREAD

**Discussion by MohanKanjula** [redouane59/twittered#444] [Q&A]
Hi , We are using this twittered client to reply to a tweet and sending a reply to DM . In our app we have want to take some  actions if there is any issue with the twitter API (ie Exception | Errors from twitter API) . 

For example :  when we are trying to reply to a tweet , If we get rate limited | we are un authorised to reply | if it's a bad request, we want to take some specific action for each type of response from Twitter API. 
But twittered client doesn't throw any exception ( i was trying out  **twitterClient.postTweet(tweetParameters)**  method is not throwing any exception )

Is there any other way to check errors | exceptions ?

**Comment by redouane59** [comment]
Good question, I think that currently nothing exists, but feel free to submit a pull request if you think you can bring something.
In the current version if you reach a limit, the code will wait for another try.
`        LOGGER.trace("Rate limit exceeded, x-rate-limit-reset: {} x-rate-limit-remaining: {}, x-rate-limit-limit:  {}", retryAfterStr, response.getHeader("x-rate-limit-remaining"), response.getHeader("x-rate-limit-limit"));` 
(See AbstractRequestHelper class)

**Reply by MohanKanjula** [reply]
Hey , Thank you for the info.

**Comment by smleggo** [comment]
How would you suggest we track when a call fails? No exceptions and doesn't look like the response is being saved anywhere. Not seeing any clean way to do it.
